File: Introduccion a POO /Libro.js
package ejercicio3;

/**
 *
 * @author belen
 */
public class Libro {
    
    private String titulo;
    private String autor;
    private int anioPublicacion;
    
    
    String getTitulo() {
        
        System.out.println("El principito");
        return titulo;
        
    }
    
    String getAutor() {
        
        System.out.println("Antoine de Saint Exupery");
        return autor;
    }
    
    
    void setAnioPublicacion(int nuevoAnio) { // validacion de año
        
        if (nuevoAnio > 1500) {
           anioPublicacion = nuevoAnio;
           System.out.print(nuevoAnio);
        
        
    } else {
            System.out.println("Año invalido.");
        }
       
}
}

File: Colecciones/Ejercicio2/Libro.java
public class Libro {
    private final String isbn;
    private final String titulo;
    private final int anioPublicacion;
    private final Autor autor;

    public Libro(String isbn, String titulo, int anioPublicacion, Autor autor) {
        this.isbn = isbn;
        this.titulo = titulo;
        this.anioPublicacion = anioPublicacion;
        this.autor = autor;
    }

    public void mostrarInfo() {
        System.out.println("Libro: " + titulo + " (" + anioPublicacion + ")");
        System.out.println("ISBN: " + isbn);
        System.out.println("Autor: " + autor);
        System.out.println("-----------------------------");
    }

    // Getters
    public String getIsbn() { return isbn; }
    public String getTitulo() { return titulo; }
    public int getAnioPublicacion() { return anioPublicacion; }
    public Autor getAutor() { return autor; }

    @Override
    public String toString() {
        return titulo + " - " + autor.getNombre();
    }
}
